test/Flick/AnimatedCarousalPage.xaml.cs

```diff
@@ -49,6 +57,76 @@ public sealed partial class AnimatedCarousalPage : Page
         public AnimatedCarousalPage()
         {
             this.InitializeComponent();
+
+            ElementCompositionPreview.GetElementVisual(sv).Clip = ElementCompositionPreview.GetScrollViewerManipulationPropertySet(sv).Compositor.CreateInsetClip();
+        }
+
+        private object selectedItem = null;
+
+        public object SelectedItem
+        {
+            get
+            {
+                return selectedItem;
+            }
+            set
+            {
+                selectedItem = value;
+            }
+        }
+
+        private bool isScrolling = false;
+
+        public bool IsScrolling
+        {
+            get
+            {
+                return isScrolling;
+            }
+        }
+
+        private double itemScaleRatio = 0.5;
+
+        public double ItemScaleRatio
+        {
+            get
+            {
+                return itemScaleRatio;
+            }
+            set
+            {
+                itemScaleRatio = value;
+            }
+        }
+
+        protected void OnScrollViewerViewChanged(object sender, ScrollViewerViewChangedEventArgs e)
+        {
+            var svCenterPoint = sv.HorizontalOffset + sv.ViewportWidth / 2;
```

Comment:
>sv.HorizontalOffset + sv.ViewportWidth / 2; [](start = 32, length = 43)

Perhaps a private helper function for this would be useful since this is used in multiple places. Need to be careful about the naming espeicially in which co-ordinate space we are talking about. maybe something like viewportCenterPointInExtent